src/main/java/org/wso2/maven/plugins/APIGenMojo.java

```diff
@@ -68,8 +74,16 @@ public void execute()
         CodegenConfig config = new CxfCodeGen();
 
         config.additionalProperties().put("invokerPackage", project.getArtifact().getArtifactId());
-        config.additionalProperties().put("apiPackage", project.getArtifact().getArtifactId());
-        config.additionalProperties().put("modelPackage", project.getArtifact().getArtifactId() + ".dto");
+        if (apiPackage != null && !apiPackage.isEmpty()) {
+            config.additionalProperties().put("apiPackage", apiPackage);
+        } else {
+            config.additionalProperties().put("apiPackage", project.getArtifact().getArtifactId());
+        }
+        if (modelPackage != null && !modelPackage.isEmpty()) {
+            config.additionalProperties().put("modelPackage", modelPackage + ".dto");
```

Comment:
Here we don't need the .dto part. This needs to be defined in the config
```suggestion
            config.additionalProperties().put("modelPackage", modelPackage);
```

Reply:
This has been fixed with https://github.com/hevayo/swagger2cxf-maven-plugin/pull/23/commits/43578578762e247b5dd88735c65080b2d4444285